js/style/style_layer.js

```diff
@@ -38,6 +38,8 @@ StyleLayer.prototype = {
                 }
                 this.layout['symbol-avoid-edges'] = true;
             }
+
+            this._resolvedLayout = new StyleDeclarationSet('layout', this.type, this.layout, this._constants || {});
```

Comment:
Yep, this isn't good. but I think we'll need to rethink some of the concepts to make this better.